addons/mail/static/src/messaging/component/notification_list/notification_list_tests.js

```diff
@@ -0,0 +1,216 @@
+        }
+        this.env = undefined;
+        delete NotificationList.env;
+    }
+});
+
+QUnit.test('notifications are ultimately ordered by date', async function (assert) {
+    assert.expect(3);
+
+    Object.assign(this.data.initMessaging, {
+        channel_slots: {
+            channel_channel: [{
+                channel_type: 'channel',
+                id: 100,
+                name: "Channel 2019",
+                message_unread_counter: 0,
+            }, {
+                channel_type: 'channel',
+                id: 200,
+                name: "Channel 2020",
+                message_unread_counter: 0,
+            }],
+        },
+    });
+    await this.start({
+        async mockRPC(route, args) {
+            if (args.method === 'channel_fetch_preview') {
+                return [
+                    {
+                        id: 100,
+                        last_message: {
+                            author_id: [10, "Author A"],
+                            body: "<p>Message A</p>",
+                            channel_ids: [100],
+                            date: "2019-01-01 00:00:00",
+                            id: 42,
+                            message_type: 'comment',
+                            model: 'mail.channel',
+                            record_name: 'Channel 2019',
+                            res_id: 100,
+                        },
+                    },
+                    {
+                        id: 200,
+                        last_message: {
+                            author_id: [20, "Author B"],
+                            body: "<p>Message B</p>",
+                            channel_ids: [200],
+                            date: "2020-01-01 00:00:00",
+                            id: 43,
+                            message_type: 'comment',
+                            model: 'mail.channel',
+                            record_name: 'Channel 2020',
+                            res_id: 200,
+                        },
+                    }
+                ];
+            }
+            return this._super(...arguments);
+        }
+    });
+    await this.createNotificationListComponent({ filter: 'all' });
+    assert.containsN(document.body, '.o_ThreadPreview', 2,
+        "there should be two thread previews");
+    const threadPreviewElList = document.querySelectorAll('.o_ThreadPreview');
+    assert.strictEqual(
+        threadPreviewElList[0].querySelector(':scope .o_ThreadPreview_name').textContent,
+        'Channel 2020',
+        "First channel in the list should be the channel of 2020 (more recent last message)"
+    );
+    assert.strictEqual(
+        threadPreviewElList[1].querySelector(':scope .o_ThreadPreview_name').textContent,
+        'Channel 2019',
+        "Second channel in the list should be the channel of 2019 (least recent last message)"
+    );
+});
+
+QUnit.test('reordering on new message received', async function (assert) {
```

Comment:
suggestion for name: "thread notifications are re-ordered on receiving a new message"